### PR TITLE
Feature: tests

### PR DESCRIPTION
**Pull request by benibr**
This PR adds some basic tests scripts that are executable via `make test`.

They test:

* connection via HTTPS to the main migrid domain
* the login+read of welcome.txt via WEBDAVS
* the login+read of welcome.txt via SFTP
* the login+read of welcome.txt via FTPS

For testing a curl docker container is used since docker must be available anyway and one can remove host dependencies this way.
There is a bug with FTPS which is triggered when curl is used with OpenSSL above `1.1.1k`. As a workaround the curl version is pinned to an older one here.

The tests work with the default.env for now and should be extended in future.
They also might serve as automatic CI testing in future to test changes in this repository.

### CODE REVIEW

File: defaults.env
Comment (jonasbardino) [inline defaults.env:1]: Was this policy change on purpose? ... it looks like a bad idea to me. The default password policy in upstream `migrid` is MEDIUM and it should not really be weaker here even for toy purposes. Choosing MEDIUM here should not interfere as long as the test user password is chosen like it is to have eight chars from three classes. One can of course still override such details while doing strictly isolated tests.

**Comment by jonasbardino** [comment]
Good idea! Merging but with manual adjustments to set the password policy back to MEDIUM for now and maybe sync to `defaults-gdp.env` in line with my comment to `defaults.env`

**Review by jonasbardino — COMMENTED** [review]
I don't understand the "correction" part here. It's an arbitrary port range and usually chosen somewhere in the range 8000-9000 for historical reasons and firewall friendliness, but dynamically negotiated with clients. We have other services running on 80XY and 8443 in some setups so we just allocated 8100-8200 to avoid conflicts. Is there any need for more passive ports?

**Comment by benibr** [comment]
> I don't understand the "correction" part here. It's an arbitrary port range and usually chosen somewhere in the range 8000-9000 for historical reasons and firewall friendliness, but dynamically negotiated with clients. We have other services running on 80XY and 8443 in some setups so we just allocated 8100-8200 to avoid conflicts. Is there any need for more passive ports?

During testing I saw failed connection attempts to FTP passive ports above 8200 so I just added the range to the env files.

I guess that https://github.com/ucphhpc/migrid-sync/blob/a9ad32683eb1b0d7a039d51c65633b6105956402/mig/shared/configuration.py#L499 somehow applies even if the `FTPS_PASSIVE_PORTS` is set, cause it is not explicitly listed in the Dockerfiles during config generation?!

File: defaults.env
Comment (benibr) [reply defaults.env:1]: Interesting, I guess I introduced this problem myself by setting the default PASSWORD_POLICY to `MODERN:12` in the Dockerfiles when I added the password policy to the generateconfs map there.

The problem was then, that the default password for the testaccount didn't meet the criterias %)

Maybe we should set all defaults to `MEDIUM` in docker-migrid in future.

Comment (jonasbardino) [reply defaults.env:1]: Yes, that's why I adjusted the default test user password to comply with the `MODERN:12` policy in the gdp env where `MEDIUM` is really too weak. I don't mind leaving the default to `MEDIUM` in the non-gdp env files. At UCPH we will just override to fit the university password policy everywhere in that case.

**Comment by jonasbardino** [comment]
> During testing I saw failed connection attempts to FTP passive ports above 8200 so I just added the range to the env files.
> 
> I guess that https://github.com/ucphhpc/migrid-sync/blob/a9ad32683eb1b0d7a039d51c65633b6105956402/mig/shared/configuration.py#L499 somehow applies even if the `FTPS_PASSIVE_PORTS` is set, cause it is not explicitly listed in the Dockerfiles during config generation?!

The corresponding `ftps_pasv_ports` setting was not exposed in `generateconfs.py`, so it was not possible to pass it there in `Dockerfile` and the environment thus had no effect either. I'll add it on the `migrid` side now so feel free to add the `docker-migrid` bindings.

**Pull request merged**